kvmagent/kvmagent/plugins/vms/tpm.py

```
@@ -0,0 +1,39 @@
+        result.type = to.type
+
+        if not check_tpm_state_vm_host_file_path_format(result.path):
+            result.error = 'invalid TPM state vm host file path ' + result.path
+            return result
+        vm_host_file.read_vm_host_file_targz(result)
+        return result
+    
+    def write_file(self, to):
+        # type: (vm_host_file.VmHostFileTO) -> None
+        if not check_tpm_state_vm_host_file_path_format(to.path):
+            raise Exception('invalid TPM state vm host file path ' + to.path)
```

Ruff (0.15.2)

[warning] Create your own exception

(TRY002)